src/modules/Builder/features/SaveAndPublish/SaveAndPublish.utils.ts

```diff
@@ -408,9 +414,12 @@ export const getActivityItems = (activity: ActivityFormValues) => {
     });
   }
 
-  return items?.map(({ id, ...item }) => ({
+  return items?.map(({ id, ...item }, index) => ({
     ...item,
     ...getItemCommonFields({ id, item, items, conditionalLogic }),
+    ...(isABTrails && {
```

Comment:
It's better to make this not on save and publish but on creating 

![image](https://github.com/ChildMindInstitute/mindlogger-admin/assets/116339462/dbd67bc7-ee40-43dd-9164-16ab75908396)